homeassistant/components/logger/__init__.py

```diff
@@ -50,33 +50,54 @@
 class HomeAssistantLogFilter(logging.Filter):
     """A log filter."""
 
-    def __init__(self, logfilter):
+    def __init__(self):
         """Initialize the filter."""
         super().__init__()
 
-        self.logfilter = logfilter
+        self._default = None
+        self._logs = None
+        self._log_rx = None
+
+    def update_default_level(self, default_level):
+        """Update the default logger level."""
+        self._default = default_level
+
+    def update_log_filter(self, logs):
+        """Rebuild the internal filter from new config."""
+        #
+        # A precompiled regex is used to avoid
+        # the overhead of a list transversal
+        #
+        # Sort to make sure the longer
+        # names are always matched first
+        # so they take precedence of the shorter names
+        # to allow for more granular settings.
+        #
+        names_by_len = sorted(list(logs.keys()), key=len, reverse=True)
```

Comment:
Copying a dict to a list copies the keys by default. We can remove `.keys()`.

Reply:
Thanks.  Fixed